cluster/addons/dashboard/dashboard-rbac.yaml

```diff
@@ -0,0 +1,49 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  labels:
+    k8s-app: kubernetes-dashboard
+    addonmanager.kubernetes.io/mode: Reconcile
+  name: kubernetes-dashboard-minimal
+  namespace: kube-system
+rules:
+  # Allow Dashboard to create 'kubernetes-dashboard-key-holder' secret.
```

Comment:
is there a reason not to create the `kubernetes-dashboard-key-holder` secret via manifest with `EnsureExists`? wouldn't that remove the need for the dashboard to create the secret (it could just get/update it)?

Reply:
In this case it should work as addon manager will manage it. Normally, without this permission if secret would not exist Dashboard would not be able to start.

Reply:
@floreks can the `kubernetes-dashboard-key-holder` be created empty via manifest, and this create permission removed? will the dashboard populate the empty secret?

Reply:
Unfortunately not. If empty secret is found then it will be deleted and during next resync period it is recreated from local copy.

Reply:
that's unfortunate. this is ok for now, but I'd like to see that switch to an update and this permission get removed in a future update. can you open an issue for that and link it here?

Reply:
What in case dashboard won't be able to recreate it and someone deletes required secret? It will crash Dashboard for good until secret is restored. This would only work in case secret is managed by addon manager.

Reply:
if it's missing, creating seems ok. if it's empty, I'd expect population via update.

Reply:
Makes sense. I have created issue for that https://github.com/kubernetes/dashboard/issues/2629